### ISSUE TITLE
Synology NAS with arm64 

### ISSUE THREAD

**Issue by Valek626** [seblucas/cops#516]
Hello,
I'm just starting to play around with my Synology, I managed to install docker and portainer, I have calibre-web running, but after installing cops I receive the error 
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address in use) 
I can't set ports with -p because I have to use docker in "host" mode, due to arm64 version limits, if I understood correctly.
So, can someone help me to set the right ports for cops? I tried to change ports in container settings but whitout success 🙄
Thank you! 🤗
Valeria

**Comment by marioscube** [comment]
You have probably fixed it by now? Or not because of docker limitations.

You could try to build COPS image yourself from a dockerfile. And set NGINX to listen on other port than 80.

As NGINX and I do not work together I cannot help you there..

**Comment by Valek626** [comment]
No, I haven't! Thank you for your help: I'll try! :)
Cheers, v.

**Comment by marioscube** [comment]
Hmmmm, if you stop calibre-web and start COPS,  do you still get that error?
COPS and calibe-web are, to an end-user, the same program (not under the hood).

Do you want an efficient utilitatian program: COPS
Do you want eye-candy that works: calibre-web

If your client is on mobile, an e-reader or on a tablet I would use COPS, if the client is on the desktop go for calibre-web.

PS: 
> I can't set ports with -p because I have to use docker in "host" mode, due to arm64 version limits

Can you point me to a source??

**Comment by Valek626** [comment]
Yes, I installed and runned cops after stopping calibre-web, I'd like to use COPS for his super fast interface, in particular from mobile device and my onyx boox readers 😊 I would use calibre-web to add ebooks from remote, mostly.

You can see some info here, for example, about synology arm limits https://wiki.servarr.com/docker-arm-synology (yes, I know, I should buy a serious NAS 😅)

Thank you! 🤗

cheers, v.

**Comment by marioscube** [comment]
Other option:

Install COPS as a website (use apache2 not NGINX unless you know what you are doing) by unzipping the zip in ../web/

If you want to use calibre-web turn off webserver. If you want to use COPS docker stop calibre-web an turn on webserver.

To ad e-books I use calibre (NOT calibre-web) on my desktop then synchronise (rsync?) or copy the databases to the server (in your case a Synology). So I always have a backup of the database and the e-books.

**Comment by Valek626** [comment]
Good idea, thank you! 👍
Yes, you're right about adding and manage books on your desktop, it would be the best solution (and suggested by calibre developer, too) but I really like editing and managing my loved ebooks on my tablet or phone... at moment I have around 11000 books 😅 so when I have time I like organizing them, maybe from the beach 😉

**Comment by marioscube** [comment]
If you have so many e-books (nice collection!) make sure you have a backup!
Finding an redownloading them will be a pain in the glutus maximux ;-)

**Comment by SylvainDx** [comment]
I was maybe in a similar situation. And I put my Calibre db on a nas share and access it from différent device (windows or linux) .... I ruin the db  several times.
Now I put à docker with the full Calibre version on the nas.  I can access it from à brosser from anywhere with my tailscale vpn. And I use cops to accès it And rénover the books

**Comment by Valek626** [comment]
thank you SylvainDx, I'll investigate! and I messed up my db several times, too 😅 (but I have a backup 😉)
cheers, v.

**Comment by SylvainDx** [comment]
For information I use : https://github.com/linuxserver/docker-calibre
you can set a funny port to this image and keep the 80 for Cops

**Comment by Valek626** [comment]
Thank you again. Yesterday I tried exactly that container, but on my NAS is terribly slow, quite unusable 🙁 maybe because I have so many books, I don't know...

**Comment by Valek626** [comment]
DS220j with a wonderful Realtek RTD1296 4-core 1.4 GHz and 512 MB DDR4 🤣 🙁 your is definitely better with Intel Celeron and 2gb RAM... wow, I'd like to compare our collections 😍 but mine is mainly in Italian...

**Comment by SylvainDx** [comment]
And mine in french ;)
Yes the 220J may be a little short for this ... at least on the memory.

**Comment by marioscube** [comment]
Well I only have an DS-110j for COPS and it works, slow.........
